Teacher Workout/Features/Authentication/SignUp/Coordinator/SignUpCoordinator.swift

```diff
@@ -2,17 +2,44 @@ import Foundation
 import SwiftUI
 import UIKit
 
+protocol SignUpCoordinatorDelegate: AnyObject {
+    func signUpCoordinatorDidClose(_ coordinator: SignUpCoordinator)
+    func signUpCoordinatorDidTapHaveAccount(_ coordinator: SignUpCoordinator)
+    func signUpCoordinatorDidTapSignUp(_ coordinator: SignUpCoordinator)
+}
+
 final class SignUpCoordinator: NSObject, Coordinator {
     var childCoordinators: [Coordinator] = []
     var navigationController: UINavigationController
 
+    weak var delegate: SignUpCoordinatorDelegate?
+
     init(navigationController: UINavigationController) {
         self.navigationController = navigationController
         super.init()
     }
 
     func start() {
-        let viewController = UIHostingController(rootView: SignUpView())
-        navigationController.pushViewController(viewController, animated: true)
+        let viewController = UIHostingController(rootView: SignUpView(delegate: self))
+        navigationController.present(viewController, animated: true)
+    }
+}
+
+extension SignUpCoordinator: SignUpViewDelegate {
+    func signUpViewDidTapClose(_: SignUpView) {
+        navigationController.dismiss(animated: true)
+        delegate?.signUpCoordinatorDidClose(self)
+    }
+
+    func signUpViewDidTapHaveAccount(_: SignUpView) {
+        navigationController.dismiss(animated: true)
+        delegate?.signUpCoordinatorDidClose(self)
+        delegate?.signUpCoordinatorDidTapHaveAccount(self)
+    }
+
+    func signUpViewDidTapSignUp(_: SignUpView) {
+        navigationController.dismiss(animated: true)
+        delegate?.signUpCoordinatorDidClose(self)
```

Comment:
@vladstanescu94 I think we should call just one delegate here, signUpCoordinatorDidTapSignUp. And the delegate to do work is needed it.